### PR TITLE
Next.js Bi-directional app

### PR DESCRIPTION
**Pull request by alejandronanez**
Example that adds a Next.js bi-directional app.

## What needs to be done

- [ ] It's not possible to load code from `app2` in `app1` as I'm getting the following error:
```
Uncaught Error: Container initialization failed as it has already been initialized with a different share scope
```
- [ ] Try out client side routing

## How to run this locally

```
# Terminal One
# In the repo root
yarn
cd nextjs-bi-directional
yarn dev
```

```
# Terminal Two
# In the repo root
cd nextjs-bi-directional/app1/sidecar
yarn build && yarn start
```

```
# Terminal Three
# In the repo root
cd nextjs-bi-directional/app2/sidecar
yarn build && yarn start
```

1. Open http://localhost:3001 
2. You should see the error in the console

## Notes
I'm using the `RemoteComponent` from `nextjs-sidecar` example. https://github.com/module-federation/module-federation-examples/blob/7dff2c4a366a17116168a38620dab8aba1dd62c3/nextjs-sidecar/host/src/pages/index.js#L4-L26

### CODE REVIEW

**Comment by alejandronanez** [comment]
I'm having some issues to get MF working between two Next.js apps. I'd appreciate if you @ScriptedAlchemy or @maraisr (or anyone really) can point me in the right direction.

Thanks a lot!

File: nextjs-bi-directional/app1/pages/index.js [hunk 1]
```diff
@@ -1,28 +1,89 @@
 import Head from "next/head";
+import Link from "next/link";
 import GreetingAppOne from "../components/GreetingAppOne";
-import { dependencies } from "../package.json";
 
-const RemoteComponent = ({ scope, module, ...props }) => {
-  if (!global[scope]) {
+const useDynamicScript = (url) => {
```
Comment (alejandronanez) [inline nextjs-bi-directional/app1/pages/index.js:5]: `useDynamicScript` and `RemoteComponent` fixed the issue I was mentioning in this comment https://github.com/module-federation/module-federation-examples/pull/217#issue-454815809 💯

**Comment by ScriptedAlchemy** [comment]
Ahhh I know how to fix this issue of container errors

File: nextjs-bi-directional/app1/pages/index.js (hunk 1 above)
Comment (alejandronanez) [reply nextjs-bi-directional/app1/pages/index.js:5]: I still don't understand why it fixed it though.

Before I was loading both remote scripts in the head of the Next.js app, once I switched to dynamic loading (`useDynamicSCript`) things started working. I'd like to understand why that happened 😅

**Comment by alejandronanez** [comment]
> Ahhh I know how to fix this issue of container errors

👋  @ScriptedAlchemy where can I read about this type of error? I've been scratching my head for a while now and I can get that to work 😅 .

Thank you!

## Update

My wild guess about what's happening here is:

1. I'm in 3001 (homepage), this will load the shell (Nav and header) which loads `remoteEntry` from `localhost:8000`
2. If I do a client side navigate to a route inside the homepage project, it will try to reload the shell, but as `remoteEntry` has already been loaded it throws.

**Pull request merged**